x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/utils.ts

```diff
@@ -211,3 +212,11 @@ export const getIndex = (getSpaceId: () => string, config: LegacyServices['confi
 
   return `${signalsIndex}-${spaceId}`;
 };
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const convertToSnakeCase = <T extends Record<string, any>>(obj: T): Partial<T> | null => {
+  return Object.keys(obj).reduce((acc, item) => {
+    const newKey = snakeCase(item);
+    return { ...acc, [newKey]: obj[item] };
+  }, {});
+};
```

Comment:
You can clean this up where you don't have that return `null` and you're not using an `any`:

```ts
export const convertToSnakeCase = <T extends Record<string, unknown>>(obj: T): Partial<T> => {
  return Object.keys(obj).reduce((acc, item) => {
    const newKey = snakeCase(item);
    return { ...acc, [newKey]: obj[item] };
  }, {});
};
```

Worked on my machine at least.